core/include/vecmem/edm/impl/host.ipp

```diff
@@ -65,6 +65,22 @@ VECMEM_HOST void host<schema<VARTYPES...>, INTERFACE>::reserve(
                                        std::index_sequence_for<VARTYPES...>{});
 }
 
+template <typename... VARTYPES, template <typename> class INTERFACE>
+VECMEM_HOST void host<schema<VARTYPES...>, INTERFACE>::push_back(
+    const object_type& element) {
+
+    // Make sure that there are some (jagged) vector types in the container.
+    static_assert(
+        std::disjunction_v<type::details::is_vector<VARTYPES>...>,
+        "This function requires at least one (jagged) vector variable.");
```

Comment:
Why does pushing back into an SoA require at least one jagged vector? :confused:

Reply:
I'm not sure I follow, one should be able to push back into an SoA of just scalar values, no?

Reply:
`vecmem::edm::type::scalar` means that we have one variable for the entire container. I.e.:

```c++
struct soa_host_container {
   int foo; // This is a scalar variable
   vecmem::vector<int>; // This is a vector variable
   vecmem::jagged_vector<int>; // This is a jagged vector
};
```

As I learned with the xAOD developments more than a decade ago, one can get easily confused about types when constantly switching between what the container has, and what the proxies have.

Reply:
Okay, I'll take your word on it.